### ISSUE TITLE
GcStress: Linux: Non deterministic failures on some tests

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/coreclr#2812]
The following tests failed (as a segfault) at least once when run under GCStress, but the failure does not repro consistently, so the bug does not seem actionable.

```
JIT/Methodical/explicit/basic/_il_relrefarg_i1/_il_relrefarg_i1
JIT/Generics/Coverage/chaos65204782cs/chaos65204782cs
JIT/jit64/opt/cse/VolatileTest_op_shr/VolatileTest_op_shr.sh
JIT/Methodical/explicit/basic/_il_relrefloc_r8/_il_relrefloc_r8.sh
JIT/Methodical/explicit/basic/_il_relrefloc_u2/_il_relrefloc_u2
Loader/regressions/polyrec/Polyrec/Polyrec
```
So, I think the task here is to re-run the following tests a few times once some of the other GcStress bugs are fixed.

The following tests are found to be passing with the fixes already in Git or the current fixes that are in PR stage:
~~JIT/Methodical/explicit/rotate/_dbgrotarg_valref/_dbgrotarg_valref.sh~~
~~JIT/Methodical/explicit/rotate/_relrotarg_valref/_relrotarg_valref.sh~~
~~JIT/Methodical/VT/callconv/_dbgjumper/_dbgjumper~~
~~JIT/Methodical/VT/callconv/_dbgjumper/_dbgjumper4~~
~~JIT/Methodical/refany/_relgcreport/_relgcreport~~
~~JIT/Methodical/refany/_relgcreport/_relgcreport~~

**Comment by sivarv** [comment]
Please verify whether any of these repro or fixed by existing fixes that we have.

**Comment by swaroop-sridhar** [comment]
I added two more tests to the above list where I saw failures at least once, but was unable to actively repro the failures.

**Comment by swaroop-sridhar** [comment]
Some of these failures could be related to #2785.
I've seen the tests occasionally fail the `!pThread->HasPendingGCStressInstructionUpdate()` assertion, but otherwise crash. 

There could be more than one problem showing up in random order.
But need to rerun the above tests once #2785 is fixed.

**Comment by sivarv** [comment]
The below test is fixed with PR #2890
JIT/Methodical/explicit/rotate/_relrotarg_valref/_relrotarg_valref.sh

**Comment by sivarv** [comment]
The below tests are passing with the fixes that @LLITCHEV is currently trying to PR
JIT/Methodical/VT/callconv/_dbgjumper/_dbgjumper
JIT/Methodical/VT/callconv/_dbgjumper/_dbgjumper4

**Comment by sivarv** [comment]
The below test is fixed with PR #2890JIT/Methodical/explicit/rotate/_dbgrotarg_valref/_dbgrotarg_valref.sh

**Comment by sivarv** [comment]
mkrefany tests that should pass with the PR @LLITCHEV  is working on.
JIT/Methodical/refany/_relgcreport/_relgcreport
JIT/Methodical/refany/_relgcreport/_relgcreport

**Comment by sivarv** [comment]
I could not repro the failure of any of the following tests

JIT/Methodical/explicit/basic/_il_relrefarg_i1/_il_relrefarg_i1
JIT/Generics/Coverage/chaos65204782cs/chaos65204782cs
JIT/jit64/opt/cse/VolatileTest_op_shr/VolatileTest_op_shr.sh
JIT/Methodical/explicit/basic/_il_relrefloc_r8/_il_relrefloc_r8.sh
JIT/Methodical/explicit/basic/_il_relrefloc_u2/_il_relrefloc_u2
Loader/regressions/polyrec/Polyrec/Polyrec

Could be possible that these might got fixed with PR #2890 or fixes to VM or disassembler.
For now closing this.  Please reopen a new bug if any of these tests fail in gc-runs after Lubo checks-in his current PR.
